lib/cfa/augeas_parser.rb

```diff
@@ -101,12 +112,28 @@ class AugeasTree
     # Low level access to Augeas structure
     #
     # An ordered mapping, represented by an Array of Hashes
-    # with the keys :key and :value.
+    # with the keys :key, :value, :operation and :orig_key.
+    #
+    # - `:key` is modified key without index used by collection
+    # - `:value` is string value or instance of AugeasTree or AugeasTreeValue
+    # - `:operation` is internal variable holding modification of augeas
+    #   structure. It is used for minimal modification of source file.
+    # - `:orig_key` is internal variable used to hold key with index
```

Comment:
yes, make sense. I always forgot about it, it is a bit non-standard ruby way for me.